etc/kapacitor/kapacitor.conf

```diff
@@ -573,27 +573,46 @@ default-retention-policy = ""
   # Use SSL but skip chain & host verification
   insecure-skip-verify = false
   ## Optional SASL Config
-  # sasl_username = "kafka"
-  # sasl_password = "secret"
+  # sasl-username = "kafka"
+  # sasl-password = "secret"
+  ## Arbitrary key value string pairs to pass as a TOML table. For example:
+  # {logicalCluster = "cluster-042", poolId = "pool-027"}
+  # sasl-extensions = {}
   ## Optional SASL:
   ## one of: OAUTHBEARER, PLAIN, SCRAM-SHA-256, SCRAM-SHA-512, GSSAPI
   ## (defaults to PLAIN)
-  # sasl_mechanism = ""
+  # sasl-mechanism = ""
   ## used if sasl_mechanism is GSSAPI
-  # sasl_gssapi_service_name = ""
+  # sasl-gssapi-service-name = ""
   # ## One of: KRB5_USER_AUTH and KRB5_KEYTAB_AUTH
-  # sasl_gssapi_auth_type = "KRB5_USER_AUTH"
-  # sasl_gssapi_kerberos_config_path = "/"
-  # sasl_gssapi_realm = "realm"
-  # sasl_gssapi_key_tab_path = ""
-  # sasl_gssapi_disable_pafxfast = false
-  ## Access token used if sasl_mechanism is OAUTHBEARER
-  # sasl_access_token = ""
-  ## Arbitrary key value string pairs to pass as a TOML table. For example:
-  # {logicalCluster = "cluster-042", poolId = "pool-027"}
-  # sasl_extensions = {}
+  # sasl-gssapi-auth-type = "KRB5_USER_AUTH"
+  # sasl-gssapi-kerberos-config-path = "/"
+  # sasl-gssapi-realm = "realm"
+  # sasl-gssapi-key-tab-path = ""
+  # sasl-gssapi-disable-pafxfast = false
+  ## Options if sasl-mechanism is OAUTHBEARER
+  ## The service name to use when authenticating with SASL/OAUTH.
+  # ## One of: "" or custom, auth0, azuread
+  # sasl-oauth-service = ""
+  ## The client ID to use when authenticating with SASL/OAUTH.
+  # sasl-oauth-client-id = ""
+  ## The client secret to use when authenticating with SASL/OAUTH.
+  # sasl-oauth-client-secret = ""
+  ## The token URL to use when authenticating with SASL/OAUTH.
+  # sasl-oauth-token-url = ""
```

Comment:
Generally, the core implementation and feature set are done according to Telegraf to provide a similar user experience
Choosing service makes sense to distinguish auth0, azure and other and provides space for future extensions

Reply:
I know as I implemented this part in Telegraf. The big issue is that the two are redundant and you need to decide **and document** the behavior if both options are given and, in the worst case, contradict each other.

Reply:
The only contradiction I see, for now, is we don't need the token-url for azure. 

I still don't understand how it would work if there were only one. The token-url?
 - we need to check if the *audience* parameter is added for auth0
 - If a user would like to use azuread, the token-url would be *azuread*?
 - In case of azure we need to check *tenant-id*

Reply:
I'm not saying you _have to_ remove it but then please make the use and redundancy clear in the option description! As it is now, people might be inclined to specify both the `token-url` _and_ the `service`.

Reply:
Sure, the token-url param should be more contextually described

Reply:
Done